[Bug]: Get Usage Bug when today is first day of the month

### 这个bug是否已存在现有issue了？

- [X] 我确认没有已有issue。

### 错误表现

```
2023-04-01 06:51:59,285 [ERROR] [openai_func.py:58] 获取API使用情况失败:API request failed with status code 400: {
"error": {
"message": "end_date must come after start_date",
"type": "invalid_request_error",
"param": null,
"code": null
}
```

### 复现操作

这个情况会在first-day == last_day -> True的时候出现
刚测试出了这个issue


### 错误日志

_No response_

### 运行环境

```markdown
- OS: 
- Browser: 
- Gradio version: 
- Python version:
```


### 补充说明

_No response_

--- a/modules/openai_func.py
+++ b/modules/openai_func.py
@@ -49,8 +49,9 @@ def get_usage(openai_api_key):
             total_used=0
             return f"**API使用情况解析失败**"
         if balance == 0:
-            last_day_of_month = datetime.datetime.now().strftime("%Y-%m-%d")
-            first_day_of_month = datetime.datetime.now().replace(day=1).strftime("%Y-%m-%d")
+            curr_time = datetime.datetime.now()
+            last_day_of_month = get_last_day_of_month(curr_time).strftime("%Y-%m-%d")
+            first_day_of_month = curr_time.replace(day=1).strftime("%Y-%m-%d")
             usage_url = f"{shared.state.usage_api_url}?start_date={first_day_of_month}&end_date={last_day_of_month}"
             try:
                 usage_data = get_billing_data(openai_api_key, usage_url)
@@ -80,3 +81,9 @@ def get_usage(openai_api_key):
     except Exception as e:
         logging.error(f"获取API使用情况失败:"+str(e))
         return standard_error_msg + error_retrieve_prompt
+
+def get_last_day_of_month(any_day):
+    # The day 28 exists in every month. 4 days later, it's always next month
+    next_month = any_day.replace(day=28) + datetime.timedelta(days=4)
+    # subtracting the number of the current day brings us back one month
+    return next_month - datetime.timedelta(days=next_month.day)

